sharding/indexHashedNodesCoordinator.go

```diff
@@ -662,8 +692,23 @@ func (ihgs *indexHashedNodesCoordinator) computeNodesConfigFromList(
 			eligibleMap[validatorInfo.ShardId] = append(eligibleMap[validatorInfo.ShardId], currentValidator)
 		case string(core.LeavingList):
 			log.Debug("leaving node trie", "pk", validatorInfo.PublicKey)
-			eligibleMap[validatorInfo.ShardId] = append(eligibleMap[validatorInfo.ShardId], currentValidator)
 			leavingMap[validatorInfo.ShardId] = append(leavingMap[validatorInfo.ShardId], currentValidator)
+			if !ihgs.flagWaitingListFix.IsSet() {
+				eligibleMap[validatorInfo.ShardId] = append(eligibleMap[validatorInfo.ShardId], currentValidator)
+			} else {
+				found, shardId := searchInMap(previousEpochConfig.eligibleMap, currentValidator.PubKey())
+				if found {
+					log.Debug("leaving node trie found in", "list", "eligible", "shardId", shardId)
+					eligibleMap[shardId] = append(eligibleMap[validatorInfo.ShardId], currentValidator)
```

Comment:
Is this correct? Why do we re-add a leaving validator to the eligible/waiting map if it was included in a previous epoch configuration as an eligible/waiting ?